connect/connect-c7/worker-adapter-c7/src/main/java/io/miragon/miranum/connect/adapter/in/c7/worker/Camunda7Mapper.java

```diff
@@ -40,7 +40,7 @@ private VariableMap toEngineData(final Map<String, Object> data) {
         final VariableMap variables = Variables.createVariables();
         data.keySet().forEach(key -> {
                     final Object value = data.get(key);
-                    if (Objects.isNull(value) || ClassUtils.isPrimitiveOrWrapper(value.getClass())) {
+                    if (Objects.isNull(value) || ClassUtils.isPrimitiveOrWrapper(value.getClass()) || value instanceof String) {
```

Comment:
This "if" looks a little bit hacky tbh. Besides the fact that it should maybe be written the other way around because that seems to be more intuitive to me at least, we may extract a variable to make it more clear what the if says. I mean I get it but, the "instanceof" does not look very robust to me

Reply:
Honestly I can't find a better way. It just did look a little bit off to me but I guess it is perfectly fine. Maybe we can just extract a variable out of the conditions to enhance readability